plugins/@grouparoo/airtable/src/lib/destination/exportRecords.ts

```diff
@@ -0,0 +1,244 @@
+  for (const record of users) {
+    inputs.push(buildUpdatePayload(record));
+  }
+  if (inputs.length == 0) {
+    return;
+  }
+  await client.updateRecords(tableId.toString(), inputs);
+};
+
+// usually this is creating them. ideally upsert. set the destinationId on each when done
+const createByForeignKeyAndSetDestinationIds: BatchMethodCreateByForeignKeyAndSetDestinationIds =
+  async (options: MethodOptions) => {
+    const { client, users, config, getByForeignKey } = options;
+    const { tableId, primaryKey } =
+      config.destinationOptions as AirtableDestinationOptions;
+    const inputs = users.map((record) => buildCreatePayload(record));
+    if (inputs.length > 0) {
+      await client
+        .createRecords(tableId.toString(), inputs)
+        .then((response) => {
+          response.map((record) => {
+            const key = record.fields[primaryKey];
+            if (!key) {
+              return;
+            }
+            const found = getByForeignKey(key.toString());
+            if (found) {
+              found.destinationId = record.id;
+            }
+          });
+        })
+        .catch((reason: AirtableError) => {
+          attachErrorsToErroneousRecords(users, reason);
+        });
+    }
+  };
+
+// fetch using the keys to set destinationId and result on BatchExports
+// use the getByForeignKey to lookup results
+const findAndSetDestinationIds: BatchMethodFindAndSetDestinationIds = async (
+  options: FindAndSetMethodOptions
+) => {
+  const { config, client, getByForeignKey, foreignKeys } = options;
+  const { tableId, primaryKey } =
+    config.destinationOptions as AirtableDestinationOptions;
+  const records = await client.listRecordsByField(
+    tableId,
+    primaryKey,
+    foreignKeys
+  );
+  records.forEach((record) => {
+    const key = record.fields[primaryKey.toString()];
+    if (!key) {
+      return;
+    }
+    const found = getByForeignKey(key.toString());
+    if (found) {
+      found.destinationId = record.id;
+      found.result = record.fields;
+    }
+  });
+};
+
+function attachErrorsToErroneousRecords(
+  records: BatchExport[],
+  error: AirtableError
+) {
+  records.map((record) => {
+    record.error = error;
+  });
+}
+
+function buildUpdatePayload(
+  exportedProfile: BatchExport
+): RecordData<Partial<FieldSet>> {
+  if (!exportedProfile.destinationId) {
+    throw ReferenceError(
+      `Could not find destination ID in Batch Export with key ${exportedProfile.recordId}`
+    );
+  }
+  return {
+    id: exportedProfile.destinationId,
+    fields: exportedProfile.newRecordProperties,
+  };
+}
+
+function buildCreatePayload(
+  exportedProfile: BatchExport
+): CreateRecord<FieldSet> {
+  return {
+    fields: exportedProfile.newRecordProperties,
+  };
+}
+
+const addToGroups: BatchMethodAddToGroups = async () => {
+  // This destination doesn't do groups
+};
+
+const removeFromGroups: BatchMethodRemoveFromGroups = async () => {
+  // This destination doesn't do groups
+};
```

Comment:
If we aren't going to use all the facilities of the batch helper, does it make sense to use it here?  cc @bleonard

Reply:
We could make it optional in `app-templates` later. This seems fine for now.